logic.py

```
@@ -762,7 +762,98 @@ def _is_candidate_valid(col):
+    # --- START FIX: Forest Plot Generation ---
+    forest_plot_html = ""
+    try:
+        # Determine which results to plot (Priority: AOR > OR)
+        plot_data = []
+        source_results = aor_results if aor_results else or_results
+        title_prefix = "Adjusted" if aor_results else "Crude"
+        
+        for label, metrics in source_results.items():
+            # Skip rows without proper data
+            if not isinstance(metrics, dict): 
+                logger.warning(f"Skipping row {label} due to invalid metrics: {metrics}")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)